File: java/Hand.java
import java.util.List;
import java.util.ArrayList;
import java.util.Collections;

public class Hand implements Comparable<Hand>
{
	public static final int NUM_CARDS_IN_HAND = 5;
	public List<Card> cardList = new ArrayList<Card>();
	private int relativeStrength = -1;

	public Hand()
	{
		
	}


	public void addCard(Card card)
	{	
		if (cardList.size() < NUM_CARDS_IN_HAND)
			cardList.add(card);
		else
			System.out.println("Error: Hand is full");
	}

	public void printHand()
	{
		for (Card card: cardList)
		{
			card.printCard();
			System.out.print("  ");
		}

		System.out.println();
	}

	public void printHandWithoutLine()
	{
		for (Card card: cardList)
		{
			card.printCard();
			System.out.print("  ");
		}
	}

	public List<Card> giveSortedCardList()
	{
		List<Card> newCardArray = new ArrayList<>(cardList);
		Collections.sort(newCardArray);
		return newCardArray;
	}

	public int getRelativeStrength()
	{
		return relativeStrength;
	}

	public void setRelativeStrength(int value)
	{
		relativeStrength = value;
	}

	//Returns 1 if other hand is stronger than this hand in contrast to regular compareTo methods
	@Override
	public int compareTo(Hand other)
	{
		return Integer.compare(other.relativeStrength, relativeStrength);
	}
}

File: c#/Hand.cs
using System.Collections.Generic;
using System;
using System.IO;

using FiveCardStud;

namespace FiveCardStud
{
public class Hand : IComparable<Hand>
{
	public const int NUM_CARDS_IN_HAND = 5;
	public List<Card> cardList = new List<Card>();
	private int relativeStrength = -1;

	public Hand()
	{
		
	}


	public void AddCard(Card card)
	{	
		if (cardList.Count < NUM_CARDS_IN_HAND)
			cardList.Add(card);
		else
			Console.WriteLine("Error: Hand is full");
	}

	public void printHand()
	{
		foreach (Card card in cardList)
		{
			card.printCard();
			Console.Write("  ");
		}

		Console.WriteLine();
	}

	public void printHandWithoutLine()
	{
		foreach (Card card in cardList)
		{
			card.printCard();
			Console.Write("  ");
		}
	}

	public List<Card> giveSortedCardList()
	{
		List<Card> newCardArray = new List<Card>(cardList);
		newCardArray.Sort();
		return newCardArray;
	}

	public int getRelativeStrength()
	{
		return relativeStrength;
	}

	public void setRelativeStrength(int value)
	{
		relativeStrength = value;
	}

	//Returns 1 if other hand is stronger than this hand in contrast to regular compareTo methods
	public int CompareTo(Hand other)
	{
		return other.relativeStrength.CompareTo(relativeStrength);
	}
}
}
